docs/internal/usage-tracker.md

```diff
@@ -0,0 +1,232 @@
+# Usage tracker service
+
+## Mission, design & decisions
+
+### Mission
+
+In Mimir's ingest architecture, the max series limit enforcement creates three critical challenges when performed by the ingesters (like in classic architecture):
+
+1. **Ingestion cost** - Series over-the-limit are ingested into Kafka/Warpstream before being rejected, generating cost for data that cannot be used.
+2. **Block-builder inconsistency** - Block-builder must independently enforce limits, potentially making different decisions than ingesters.
+3. **Client error feedback** - Errors are not reported back to clients in HTTP responses.
+
+Usage tracker service solves these by enforcing per-tenant max series limits before Kafka ingestion.
+
+Usage tracker is designed to be a highly available, horizontally scalable service that tracks series hashes and enforces series limits across multiple availability zones.
+
+The synchronization between multiple availability-zones is eventually consistent, and it leverages a Kafka-compatible storage to both synchronize replicas in runtime and to persist the state of the service upon restarts.
+
+### Consistency over precision
+
+This service is not designed to be meant as a cost-control mechanism but a system protection.
+
+The consistency of decisions is considered more important than the precision of the limiting algorithm,
+so in some scenarios the system will accept more series than the configured limit, but it will reject series that were already accepted and are still active.
```

Comment:
This last bit doesn't make sense to me. It may accept more than the limit, but it will reject series that were already accepted?